Implemented IOConnector for NeulandSim/Digi

SimonL2718 opened the pull request:
Describe your proposal.

Mention any issue this PR is resolved or is related to.

---

Checklist:

* [ ] Rebased against `dev` branch
* [ ] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [ ] Followed the [pull request guidlines](https://opensource.creativecommons.org/contributing-code/pr-guidelines/) and the [Git workflow](https://github.com/AnarManafov/GitWorkflow/blob/master/GitWorkflow.markdown)
* [ ] Followed the [seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

YanzhaoW committed:
Find ROOT using config mode and cmake targets

CMake targets imported from ROOT using config mode is suggested both
from ROOT and FairRoot. Libraries generated with different detectors are
also represented by proper CMake targets for easier dependency
management.

hapol committed:
Include CEPA in geometry generation macro

Add new parameter container for FEBEX PC

Some corrections in the CALIFA geometry macro are also included.

Modify number of crystals and correct bug in geometry creation macro

The number of crystals is now 2544, including CEPA.

Include support for CEPA crystals and 2024.2 geometry

Clang-formatted, including correction in clang-format configuration file

Solve undetermined variable and clang-format

Also enables automatic califa geo generation

jose-luis-rs committed:
Update eventheader

Update eventheader

Update

Minor change

Removed unused file califa/geobase/create_califa_geo.cxx

Minor change

Minor change

Minor change

jose-luis-rs committed:
Minor update for CALIFA simulation classes

Minor change

Added include TArray.h in calibration/R3BCalifaMapped2CrystalCalPar.h

Added include TArray.h in calibration/R3BCalifaMapped2CrystalCalPar.h

Added include TArrayF.h in calibration/R3BCalifaMapped2CrystalCalPar.h

Added include TArrayF.h in calibration/R3BCalifaMapped2CrystalCal.h

Update R3BCalifa class for simulation and related classes

Minor change

Minor change in r3bcalifa

Check compilation

Almost ready the new implementation

Minor change

Minor update

Added final IOConnector for output data

Added final IOConnector for output data

Minor changes

Clang-format

Clang-format

Minor change

Minor change

Minor change

Minor change

Minor change

Minor change

Minor change

Minor change in R3BCalifaCrystalCal2Cluster.cxx

Use of TObject for R3BCalifaClusterData

Minor change

Improved version for Califa digitizer

Last modifications

Final update

Last update

Removing unused variables

Update cmake policies

New data structures for Point, CAL and Hit data

Update R3BCalifaMappedData

Removed factor 1000 in the digitizeer, found by Gabriel

Update the calculations of califa resolutions

audreychatillon committed:
Create FrssciMappedData
Create r3BFrssciReader
clang-format frssci files

clang-format-15 -i for frssci files

debug tf / tc mismatch

remove auto item from the reader

Add online spectra for Frssci at MAPPED LEVEL

coherent type of variables

cosmetics

clang

cosmetics

cosmetics

fix bug

comment fh2_multMap_RvsL[i]->Fill( to try to pass the ckang-tidy test

try to fix the clang-tidy requirement

R3BFrsSci Mapped level small changes

debug ext_h101

create MAPPED LEVEL for FrsSci

Starting to implement the template for ttt10

Some clases to move data from mapped to cal

Added containers

clang-format

Update tttx reader

Removed unused line

clang-format

Update

Minor change

TTTX: Added Cal2Hit and HitPar classes

TTTX: Added more calibration parameters for CAL and HIT levels

Minor change

Minor change

Clang-format for ext_h101_frssci.h

jose-luis-rs committed:
Update of R3BCalifaOnlineSpectra

Use of std::string in R3BCalifaOnlineSpectra instead of sprintf

Update definition of histograms

Update califa online class

Update string side

Update ring value for mapped histograms

Removed inline

Basic changes

jose-luis-rs committed:
Update ALPIDE data structures

Minor change

igasparic committed:
correct neuland ext header file for s118 unpacker

audreychatillon committed:
create MAPPED LEVEL for FrsSci

audreychatillon committed:
fix compilation for gcc > 8

audreychatillon committed:
add FrsSci task to fill Tcal level

debug a copy/paste bug

fix small bugs

debug

debug

reoder init

change som type, to stay coherent

debug

try to understand the status if clang-tidy :´(

another try to understand the message from clang-tidy

still trying to find out some messages of clang-tidy

cannot call virtual function in destructor ?

forget to add a class in CMakeLists

additional spectra for the Tcal level

forget R3BFrsSciMapped2TcalPar in LinkDef

typo fixed

jose-luis-rs committed:
Created status badge

Update

Update2

Update3

Update4

Update5

Update6

Update7

Update8

Update9

mwhiteheadyork committed:
Create R3BTttxOnlineSpectra

ryotani committed:
R3BTttxMapped2Cal is added

TTTXReader is renamed to TttxReader
Mapped2Cal is updated to reconstruct the calibrated energy and the relative timing to trigger.
The initial values for Mwpc hit postion is set to NAN.

jose-luis-rs committed:
Fibers: Added Enis online task

Update

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

Update for clang-tidy

ManuelXarepe committed:
updated classes for s091/s118 part1

audreychatillon committed:
debug Tcalibrator while running the macro

jose-luis-rs committed:
FRSSCI: Minor changes

ryotani committed:
R3BTttxCal2Hit and others

Now CalData will be sorted with a descending order of energy first, then it will be clustered through the Cal2Hit process based on the time and geometrical distances between the hits.
The clusterd hits for each detector will be again checked with the time-correlations to the other detector, then reconstruct the Z.
The hit parameters are defined detector-by-detector to accomodate different thicknesses. The weights to average the values are half and half.
The HitData will include the DetectorID to indicate how reconstruction is made. If two detectors are used, DetID=0 will be assigned, otherwise, if only one detector is used, the actual DetID will be recorded.

jose-luis-rs committed:
FOOT: Update classes

More update for parameter containers

update FOOT classes

FOOT:Update online class

Clang-format

Minor changes

jose-luis-rs committed:
MWPC: Update online

Minor change

jose-luis-rs committed:
TWIM:Update online and others

Minor change

Minor change

Minor change

Minor change

Minor change

Minor change

jose-luis-rs committed:
Spider:Added online for correlations with Califa

Added histograms

Minor change

jose-luis-rs committed:
Added online for correlations between mwpc and tttx

Minor change

Added histograms for multiplicity correlations

jose-luis-rs committed:
Fixing the R3BIncomingTrackingOnlineSpectra for 2024 experiments

Added the right calculation for the incoming tracking of 2024

audreychatillon committed:
FrsSci: PosCalData + TofCalData + CalPar + Tcal2Par

try to solve warning from clang

debug

audreychatillon committed:
Add online spectra for R3BFrsSciCalData

audreychatillon committed:
reduce the number of R3BFrsSciCalPar and change some names

debug

audreychatillon committed:
missing R3BFrsSciTcal2Cal in CMakeLists

minor change from macro reading with pulser

minor debug ContFact

cosmetics

cosmetics

cosmetics

cosmetics

tmp: some print to understand problem in execution of the macro

remove debug info

cosmetics

debug online for CAL level while running macro

some debugging while running macro

debug FrsSciTcal2Cal in case no hit found

remove french comment

try to fix clang-tidy

ifndef -> pragma once

fix problem of array size for some parameters

jose-luis-rs committed:
New data structure for tofd

Minor change

audreychatillon committed:
Add AoQ at TofCalData

update FrsSciCalPar with BRho0 and dispersion

fill TofCal data with BRho (not gamma) and AoQ

add online spectra for BRho and A/Q

debug typo during compilation

fix bugs while running make

again

audreychatillon committed:
Prepare online spectra for FrsSci data versus Musli data

still on-going

2D histos to get on-line correlation between FrsSciCal vs (MusliCal and MusliHit)

debug thanks to clang-tidy

minor

stupid bug

jose-luis-rs committed:
Update tofd and fiber classes for 2024 beam time

jose-luis-rs committed:
Update ALPIDE classes

Minor change

audreychatillon committed:
fix Online frssci Vs Musli with macro

jose-luis-rs committed:
TOFD: Added histograms to display the Y position per plane at CAL level

Minor change

audreychatillon committed:
add comments in R3BMusliMapped2Cal

clang-format

change some variable names to get difference between time and drift time

cosmetics

cosmetics

remove one #define to get the value of the params

back to previous version with #define

 change ifndef to pragma once

minor changes

stupid bug

jose-luis-rs committed:
Minor change

hapol committed:
New ext_h101_califa.h file for CALIFA with extended number of crystals

Add clang-format

jose-luis-rs committed:
General update

Clang-format

Clang-format again

Clang-format

Clang-format

jose-luis-rs committed:
Added data structure for mosaic boards

clang-format

jose-luis-rs committed:
Update rolu and tcal classes

Update cleanup_cache.yml

Tests

jose-luis-rs committed:
Create cleanup_cache.yml

jose-luis-rs committed:
Update cleanup_cache.yml to v4

jose-luis-rs committed:
Update github checkout actions to version 4

jose-luis-rs committed:
Update readers

jose-luis-rs committed:
TOFD: Added time difference with respect trigger times for each PMT

New script to cleanup the cache

jose-luis-rs committed:
Cleanup only the largest caches

jose-luis-rs committed:
Update r3bsource/base classes and added a setter to R3BUcesbSource for selecting full ucesb data structure validation

Minor change

jose-luis-rs committed:
LOS: Added trailing trigger times to the reader

jose-luis-rs committed:
ALPIDE: Added new reader for Mosaics

Minor change

Solved problem with R3BTrloiiTpatReader.h

Clang-format

Commit:
Implement particle filtering via NeulandPointFilter

audreychatillon committed:
minor changes in R3BMusliMap2Cal to try to get read of the define

debug

on-going to get rid of the define hard code

on-going

try to understand the clang-check warning

really I do not understand the warning

:'(

really do not see why map_eraw[i][j] and map_traw[i][j] are not defined

try something else 1D array instead of 2D

painful

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.cxx:
```diff
@@ -64,48 +65,48 @@ void R3BNeulandDigitizer::SetParContainers()
     std::map<UInt_t, Double_t> paddleEnergyDeposit;
     // Look at each Land Point, if it deposited energy in the scintillator, store it with reference to the bar
-    for (const auto& point : fPoints.Retrieve())
+    for (const auto& point : neuland_points_.get())
```
```suggestion
    for (const auto& point : neuland_points_)
```

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.cxx:
```diff
@@ -26,28 +26,29 @@
-    , fPoints(std::move(input))
-    , fHits(std::move(output))
-    , fDigitizingEngine(std::move(engine))
+    , neuland_points_(input)
+    , neuland_hits_(output)
```
```suggestion
```
Don't need input and output here.

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.cxx:
```diff
@@ -26,28 +26,29 @@
     : R3BNeulandDigitizer(Digitizing::CreateEngine(UsePaddle<NeulandPaddle>(), UseChannel<TacquilaChannel>()),
-                          std::move(input),
-                          std::move(output))
+                          input,
+                          output)
```
Same below.

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.cxx:
```diff
@@ -26,28 +26,29 @@
 #include <utility>
 
-R3BNeulandDigitizer::R3BNeulandDigitizer(TString input, TString output)
+R3BNeulandDigitizer::R3BNeulandDigitizer(std::string_view input, std::string_view output)
```
```suggestion
R3BNeulandDigitizer::R3BNeulandDigitizer()
```

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.cxx:
```diff
@@ -64,48 +65,48 @@ void R3BNeulandDigitizer::SetParContainers()
     auto const PaddleMulSize = 3000;
-    hMultOne = R3B::root_owned<TH1I>(
+    mult_one_ = R3B::root_owned<TH1I>(
         "MultiplicityOne", "Paddle multiplicity: only one PMT per paddle", PaddleMulSize, 0, PaddleMulSize);
-    hMultTwo = R3B::root_owned<TH1I>(
+    mult_two_ = R3B::root_owned<TH1I>(
         "MultiplicityTwo", "Paddle multiplicity: both PMTs of a paddle", PaddleMulSize, 0, PaddleMulSize);
     auto const timeBinSize = 200;
-    hRLTimeToTrig = R3B::root_owned<TH1F>("hRLTimeToTrig", "R/Ltime-triggerTime", timeBinSize, -100., 100.);
+    rl_time_to_trig_ = R3B::root_owned<TH1F>("hRLTimeToTrig", "R/Ltime-triggerTime", timeBinSize, -100., 100.);
```
Use R3B::DataMonitor

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.h:
```diff
@@ -60,42 +61,42 @@ class R3BNeulandDigitizer : public FairTask
+                                 std::string_view input = "NeulandPoints",
+                                 std::string_view output = "NeulandHits");
 
     ~R3BNeulandDigitizer() override = default;
```
```suggestion
```

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.h:
```diff
@@ -60,42 +61,42 @@ class R3BNeulandDigitizer : public FairTask
-    R3BNeulandDigitizer(R3BNeulandDigitizer&&) = delete;                 // move constructor
-    R3BNeulandDigitizer& operator=(const R3BNeulandDigitizer&) = delete; // copy assignment
-    R3BNeulandDigitizer& operator=(R3BNeulandDigitizer&&) = delete;      // move assignment
+    R3BNeulandDigitizer(const R3BNeulandDigitizer&) = delete;                    // copy constructor
+    R3BNeulandDigitizer(R3BNeulandDigitizer&&) = delete;                         // move constructor
+    auto operator=(const R3BNeulandDigitizer&) -> R3BNeulandDigitizer& = delete; // copy assignment
+    auto operator=(R3BNeulandDigitizer&&) -> R3BNeulandDigitizer& = delete;      // move assignment
```
```suggestion
```

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.cxx:
```diff
@@ -26,38 +26,55 @@
 #include <utility>
 
-R3BNeulandHitMon::R3BNeulandHitMon(TString input, TString output, const Option_t* option)
+R3BNeulandHitMon::R3BNeulandHitMon(std::string_view input, std::string_view output, const Option_t* option)
```
Save above

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.h:
```diff
@@ -35,55 +37,56 @@ class TH1I;
+                              std::string_view output = "NeulandHitMon",
                               const Option_t* option = "");
 
     ~R3BNeulandHitMon() override = default;
```
```suggestion
```

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.h:
```diff
@@ -35,55 +37,56 @@ class TH1I;
-    R3BNeulandHitMon(R3BNeulandHitMon&&) = delete;                 // move constructor
-    R3BNeulandHitMon& operator=(const R3BNeulandHitMon&) = delete; // copy assignment
-    R3BNeulandHitMon& operator=(R3BNeulandHitMon&&) = delete;      // move assignment
+    R3BNeulandHitMon(const R3BNeulandHitMon&) = delete;                    // copy constructor
+    R3BNeulandHitMon(R3BNeulandHitMon&&) = delete;                         // move constructor
+    auto operator=(const R3BNeulandHitMon&) -> R3BNeulandHitMon& = delete; // copy assignment
+    auto operator=(R3BNeulandHitMon&&) -> R3BNeulandHitMon& = delete;      // move assignment
```
```suggestion
```

jose-luis-rs committed:
New ranges for TofD histograms done by Leyla and some updates for Fibers done by Valerii

Minor change

jose-luis-rs committed:
Added new histograms for califa PID

ryotani committed:
Hot fix for the TTTX classes

jose-luis-rs committed:
Minor changes for TofD

jose-luis-rs committed:
Fix some mistakes in tofd classes

Minor change

ryotani committed:
Hot fix tttx2. Channel ID is 1-oriented

jose-luis-rs committed:
Update version for R3BTofDHitModulePar.h

jose-luis-rs committed:
Added online for the TofD and Tttx charge correlation

Minor change

Minor change

Minor change

audreychatillon committed:
some debug

Adapt MusliCal2Hit for FrsSciCal data

debug the part of Cal2Hit using FrsSciTofCal

mwhiteheadyork committed:
Including Cal and Hit levels to TTT10 Online

audreychatillon committed:
fix stupid bug of index out of range

remove some useless LOG(info)

jose-luis-rs committed:
Changed algorithm to correlated tofd and tttx charges

audreychatillon committed:
debug MusliCal2Hit calibration and FrsSciVsMusli online

jose-luis-rs committed:
Updates for tttx onlines

jose-luis-rs committed:
Minor change for califa

Another minor change for califa

igasparic committed:
added trigger time trailing edge in neuland tamex

clang-format done

audreychatillon committed:
Create online for Mwpc@Cal vs Musli@Cal

Mw vs Musli for Xup and Xdown

jose-luis-rs committed:
Minor updates for R3BMwpcCorrelationOnlineSpectra and R3BMosaicReade

jose-luis-rs committed:
Update R3BNeulandOnlineSpectra and R3BNeulandTamexReade classes

Minor changes

jose-luis-rs committed:
Update califa online by Martina

Minor change

jose-luis-rs committed:
Added basic mapping for ALPIDE sensors during the commissioning

mwhiteheadyork committed:
Fixes for TTT10 Online

audreychatillon committed:
update online for Musli and FrsSci

YanzhaoW commented on neuland/digitizing/R3BNeulandDigitizer.h:
```diff
@@ -61,18 +61,8 @@ class R3BNeulandDigitizer : public FairTask
-    R3BNeulandDigitizer(R3BNeulandDigitizer&&) = delete;                         // move constructor
-    auto operator=(const R3BNeulandDigitizer&) -> R3BNeulandDigitizer& = delete; // copy assignment
-    auto operator=(R3BNeulandDigitizer&&) -> R3BNeulandDigitizer& = delete;      // move assignment
+    explicit R3BNeulandDigitizer();
```
No need for explicit. Check [this](https://en.cppreference.com/w/cpp/language/explicit) about when to use explicit.

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.h:
```diff
@@ -35,55 +37,46 @@ class TH1I;
-    TH1D* hY = nullptr;
+    std::string output_;
+
+    R3B::InputVectorConnector<R3BNeulandHit> neuland_hits_;
```
Specify the branch name here, not in the constructor

```suggestion
    R3B::InputVectorConnector<R3BNeulandHit> neuland_hits_ {"NeulandHits"};
```

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.h:
```diff
@@ -35,55 +37,46 @@ class TH1I;
-    TH1D* hE = nullptr;
-    TH1D* hX = nullptr;
-    TH1D* hY = nullptr;
+    std::string output_;
```
Prefer initializing member variables here, instead of in the constructor.

```suggestion
    std::string output_ {"NeulandHitMon"};
```

YanzhaoW commented on neuland/digitizing/R3BNeulandHitMon.cxx:
```diff
@@ -26,38 +26,55 @@
     : FairTask("R3B NeuLAND NeulandHit Monitor")
-    , fOutput(std::move(output))
-    , fHits(std::move(input))
+    , output_("NeulandHitMon")
+    , neuland_hits_("NeulandHits")
```
```suggestion

```

YanzhaoW reviewed (changes requested):
Thanks for the change. Here are some further suggestions.

jose-luis-rs committed:
Improving ALPIDE classes

Added more histograms for alpide correlations

Clang-format

jose-luis-rs committed:
Minor changes for fibers and tofd

jose-luis-rs committed:
Added more options for tracking at cave entrance

jose-luis-rs committed:
Update clases from beam time

jose-luis-rs committed:
Alpide: Minor change

Minor changes for online

MartinaFeijoo committed:
califa online update

MartinaFeijoo committed:
califa online clang format

YanzhaoW commented:
Good, for the next, you could start to use the class DataMonitor for histograms.

jose-luis-rs committed:
Added to the eventheader a LOS time without trigger times

Minor change

Minor change

Minor change

jose-luis-rs committed:
Update classes from the beam time

Minor change for fibers

Clang-format

Small mistake

ryotani committed:
2D correlation plot of tpat in R3BGeneralOnlineSpectra

A 2D plot to illustrate tpat values in each event is added. This will be useful to monitor the Trloii functions for Foot-on and Foot-bonus triggers.

audreychatillon committed:
TofRaw(FrsSci -> Los) @FrsSciTcal level, if mult=1

debug the FrsSciVsLos while compiling

add a setter for LOS pmt id used as Tref

jose-luis-rs committed:
Changes from beam time

Update R3BScalersOnlineSpectra and R3BTofDOnlineSpectra

Clang-format

Minor change

Clang-format

Clang-format

Clang-format

jose-luis-rs committed:
GLAD:Minor modifications to solve problems with neuLand simulations

YanzhaoW committed:
digitization tamex improvement

YanzhaoW committed:
fix seg fault

YanzhaoW committed:
fix ctest

ryotani committed:
Updated IncomingBeta reconstruction for s091 experiment.

Clean-up and minor improvements in los classes.

Include timing of S2 hit after subtraction of the reference time in FrsSciPosCalData.

Bug fix in R3BFrsSciTcal2Cal when obtaining good pairs of hits of left and right PMTs.

Update IncomingData to reconstruct S2(FrsSci) to LosCal for s091 experiment. Los CFD out was used for the Tref signal for S2, so LosCal is used for reconstructing ToF specific to the experiment.
When multiple hits in Los are seen, only selecting good hit wrt the trigger timing is choosen.

Morningrize committed:
Revert naming scheme for trigger leading time

YanzhaoW commented on neuland/digitizing/NeulandPointFilter.cxx:
```diff
@@ -2,45 +2,68 @@
+
+auto NeulandPointFilter::check_pid_for_criteria(int pid, BitsetParticle criteria) -> bool
+{
+    return ((CheckCriteria(BitsetParticle::proton, criteria) && pid == proton_pid) ||
+            (CheckCriteria(BitsetParticle::neutron, criteria) && pid == neutron_pid) ||
+            (CheckCriteria(BitsetParticle::electron, criteria) && pid == electron_pid) ||
+            (CheckCriteria(BitsetParticle::positron, criteria) && pid == positron_pid) ||
+            (CheckCriteria(BitsetParticle::alpha, criteria) && pid == alpha_pid) ||
+            (CheckCriteria(BitsetParticle::gamma, criteria) && pid == gamma_pid));
+}
```
You should write another conversion function that converts the PID to BitsetParticle.

I could think of two ways:
1. Use std::unordered_map
```cpp
auto PID_to_BitSetParticle = std::unordered_map<int, BitsetParticle>{{2212, BitsetParticle::proton}, {2112, BitsetParticle::neutron}};
```
And then in the conversion function, you get the particle from its PID with
```cpp
auto particle = PID_to_BitSetParticle.at(pid);
```
2. Use switch case:
```cpp
switch(pid){
case 2212:
    return BitsetParticle::proton; break;
case 2112:
    return BitsetParticle::neutron;break;
}
```

Morningrize committed:
Clang-tidy fix and suppress

jose-luis-rs committed:
Changed the cleanup of cache to 2 months

Update version v3 to v4

Removed unused includes

YanzhaoW committed:
improve template of task class

jose-luis-rs committed:
Update for compilation with FairSoft jan24

Clang-format

jose-luis-rs committed:
Update .github/workflows/cleanup_cache.yml

YanzhaoW committed:
Print the value of std::error_code instead

YanzhaoW committed:
fix a small error in set_ld_lirbary_dir

YanzhaoW committed:
Add whitelist and data output to R3BPhaseSpaceGen

Fix the value of the total energy

YanzhaoW committed:
Fix frs and asyeos cmake errors

YanzhaoW committed:
Fix epics cmake target issues

YanzhaoW committed:
integrate conan package manager

Conan package manager is now added to R3BRoot. By default, it's disable.
Building the project with conan package manager can be achieved with
cmake preset and conan cmake wrapper as a git submodule.

YanzhaoW committed:
Add data monitor and ValueError and update on filesource

Adding new class R3BDataMonitor for easy histogram operations in
FairTask. New class ValueError can be used to represent values with
errors. R3BFileSource2 now can handle root files just with root tree
inside. Documentation of these new classes and changes have been added
to the README.md in the r3bbase folder.

YanzhaoW committed:
remove REQUIRED from GTest
